src/snowcli/cli/nativeapp/commands.py

```diff
@@ -36,8 +36,20 @@
 @with_output
 @global_options
 def app_init(
-    name: str = typer.Argument(
-        ..., help="Name of the Native Apps project to be initiated."
+    path: str = typer.Argument(
+        ...,
+        help=f"""Directory to be initialized with the Native Application project. This directory must not already
+        exist.""",
+    ),
+    name: str = typer.Option(
+        None,
+        help=f"""The name of the native application. This defaults to the name of the directory when it is not specified
+        explicitly. When defaulting the application name, any group of of '.', '-' or ' ' will be replaced with '_'. The
+        name will be treated as a regular unquoted identifier if allowed by Snowflake's identifier syntax, and as a
+        quoted identifier otherwise. The name can be explicitly quoted by including the surrounding double quotes in
+        the name. See https://docs.snowflake.com/en/sql-reference/identifiers-syntax for more details on identifiers.
+        The name of the application can also be changed in the generated snowflake.yml if desired.
```

Comment:
Nit: this name is technically the name of the application _project_, not the application itself. But I'm not sure being precise about the distinction is worth the confusion it might cause users.